### ISSUE TITLE
What wireless network card supports AP mode

### ISSUE THREAD

**Issue by caixiawang** [sophron/wifiphisher#221]
Give me some advice. Thank you.

**Comment by blackHatMonkey** [comment]
Hi @caixiawang 
I recommend getting the ```Alfa AWUS036NHA``` which is confirmed to work. You can either get two ```Alfa AWUS036NHA``` or one ```Alfa AWUS036NHA``` and any wireless adapter that supports monitor mode.

**Comment by caixiawang** [comment]
Thank you very much.

**Comment by caixiawang** [comment]
@blackHatMonkey 
@strasharo 
No place to buy AWUS036NHA and TL-WN722N.
Are there any other alternatives?

**Comment by caixiawang** [comment]
I used two rtl8187 and TL-WN722N card, do not create false ap? Please advise

**Comment by blackHatMonkey** [comment]
Did you get any errors?
You also can try running ```wifiphisher``` using the ```-aI``` argument like so:
```bash
sudo python wifiphisher -aI wlan0
```

**Comment by caixiawang** [comment]
I do not speak English, please forgive me

**Comment by caixiawang** [comment]
Ask the teacher, my two NIC can support?

**Comment by caixiawang** [comment]
Thank you teacher to help me, with the command sudo python wifiphisher.py -aI wlan0 run wifiphisher, still can not create a false ap

**Comment by caixiawang** [comment]
root@kali:~# ifconfig -a
eth0      Link encap:Ethernet  HWaddr 00:0c:29:e9:57:f8  
          inet addr:192.168.44.128  Bcast:192.168.44.255  Mask:255.255.255.0
          inet6 addr: fe80::20c:29ff:fee9:57f8/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:149 errors:0 dropped:0 overruns:0 frame:0
          TX packets:83 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:17886 (17.4 KiB)  TX bytes:12395 (12.1 KiB)
          Interrupt:19 Base address:0x2000 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:24 errors:0 dropped:0 overruns:0 frame:0
          TX packets:24 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:1440 (1.4 KiB)  TX bytes:1440 (1.4 KiB)

wlan0     Link encap:UNSPEC  HWaddr 00-E0-6C-81-E5-80-3A-30-00-00-00-00-00-00-00-00  
          inet addr:10.0.0.1  Bcast:10.0.0.255  Mask:255.255.255.0
          UP BROADCAST MULTICAST  MTU:1400  Metric:1
          RX packets:2330 errors:0 dropped:1306 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:398677 (389.3 KiB)  TX bytes:0 (0.0 B)

wlan2     Link encap:Ethernet  HWaddr 3c:46:d8:14:0b:c8  
          inet addr:10.0.0.1  Bcast:10.0.0.255  Mask:255.255.255.0
          UP BROADCAST MULTICAST  MTU:1400  Metric:1
          RX packets:940 errors:0 dropped:370 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:168465 (164.5 KiB)  TX bytes:0 (0.0 B)

**Comment by caixiawang** [comment]
How to TP-link TL-wn722n set AP mode

**Comment by caixiawang** [comment]
TP-link TL-wn722n how to set up the AP model

**Comment by caixiawang** [comment]
TP-link TL-wn722n how to set up the AP mode

**Comment by caixiawang** [comment]
root@kali:~# airmon-ng

PHY	Interface	Driver		Chipset

phy0	wlan0		rtl8187		Realtek Semiconductor Corp. RTL8187
phy1	wlan1		ath9k_htc	Atheros Communications, Inc. AR9271 802.11n

root@kali:~# 
This is my two NIC

**Comment by blackHatMonkey** [comment]
Can you post the output of these commands
```bash
iw phy phy0 info
iw phy phy1 info
```

**Comment by strasharo** [comment]
According to your airmon output wlan1 is your TL-wn722n adapter. Can you try running instead:
sudo python wifiphisher.py -aI wlan1 -jI wlan0. I have the same pair of cards and it works.

**Comment by caixiawang** [comment]
@blackHatMonkey 
root@kali:~# iw phy phy0 info
Wiphy phy0
	max # scan SSIDs: 4
	max scan IEs length: 2285 bytes
	Retry short limit: 7
	Retry long limit: 4
	Coverage class: 0 (up to 0m)
	Device supports RSN-IBSS.
	Supported Ciphers:
		* WEP40 (00-0f-ac:1)
		* WEP104 (00-0f-ac:5)
		* TKIP (00-0f-ac:2)
		* CCMP (00-0f-ac:4)
		* 00-0f-ac:10
		* GCMP (00-0f-ac:8)
		* 00-0f-ac:9
	Available Antennas: TX 0 RX 0
	Supported interface modes:
		 * IBSS
		 * managed
		 * monitor
	Band 1:
		Bitrates (non-HT):
			* 1.0 Mbps
			* 2.0 Mbps
			* 5.5 Mbps
			* 11.0 Mbps
			* 6.0 Mbps
			* 9.0 Mbps
			* 12.0 Mbps
			* 18.0 Mbps
			* 24.0 Mbps
			* 36.0 Mbps
			* 48.0 Mbps
			* 54.0 Mbps
		Frequencies:
			* 2412 MHz [1] (20.0 dBm)
			* 2417 MHz [2] (20.0 dBm)
			* 2422 MHz [3] (20.0 dBm)
			* 2427 MHz [4] (20.0 dBm)
			* 2432 MHz [5] (20.0 dBm)
			* 2437 MHz [6] (20.0 dBm)
			* 2442 MHz [7] (20.0 dBm)
			* 2447 MHz [8] (20.0 dBm)
			* 2452 MHz [9] (20.0 dBm)
			* 2457 MHz [10] (20.0 dBm)
			* 2462 MHz [11] (20.0 dBm)
			* 2467 MHz [12] (20.0 dBm)
			* 2472 MHz [13] (20.0 dBm)
			* 2484 MHz [14] (disabled)
	Supported commands:
		 * new_interface
		 * set_interface
		 * new_key
		 * start_ap
		 * new_station
		 * new_mpath
		 * set_mesh_config
		 * set_bss
		 * authenticate
		 * associate
		 * deauthenticate
		 * disassociate
		 * join_ibss
		 * join_mesh
		 * set_tx_bitrate_mask
		 * frame
		 * frame_wait_cancel
		 * set_wiphy_netns
		 * set_channel
		 * set_wds_peer
		 * probe_client
		 * set_noack_map
		 * register_beacons
		 * start_p2p_device
		 * set_mcast_rate
		 * Unknown command (104)
		 * connect
		 * disconnect
	Supported TX frame types:
		 * IBSS: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * managed: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * AP: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * AP/VLAN: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * mesh point: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-client: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-GO: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-device: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
	Supported RX frame types:
		 * IBSS: 0x40 0xb0 0xc0 0xd0
		 * managed: 0x40 0xd0
		 * AP: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * AP/VLAN: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * mesh point: 0xb0 0xc0 0xd0
		 * P2P-client: 0x40 0xd0
		 * P2P-GO: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * P2P-device: 0x40 0xd0
	software interface modes (can always be added):
		 * monitor
	interface combinations are not supported
	HT Capability overrides:
		 * MCS: ff ff ff ff ff ff ff ff ff ff
		 * maximum A-MSDU length
		 * supported channel width
		 * short GI for 40 MHz
		 * max A-MPDU length exponent
		 * min MPDU start spacing
	Device supports TX status socket option.
	Device supports HT-IBSS.
	Device supports SAE with AUTHENTICATE command
	Device supports low priority scan.
	Device supports scan flush.
	Device supports AP scan.
	Device supports per-vif TX power setting
	Driver supports a userspace MPM

**Comment by caixiawang** [comment]
root@kali:~# iw phy phy1 info
Wiphy phy1
	max # scan SSIDs: 4
	max scan IEs length: 2257 bytes
	Retry short limit: 7
	Retry long limit: 4
	Coverage class: 0 (up to 0m)
	Device supports RSN-IBSS.
	Device supports T-DLS.
	Supported Ciphers:
		* WEP40 (00-0f-ac:1)
		* WEP104 (00-0f-ac:5)
		* TKIP (00-0f-ac:2)
		* CCMP (00-0f-ac:4)
		* 00-0f-ac:10
		* GCMP (00-0f-ac:8)
		* 00-0f-ac:9
		* CMAC (00-0f-ac:6)
		* 00-0f-ac:13
		* 00-0f-ac:11
		* 00-0f-ac:12
	Available Antennas: TX 0x1 RX 0x1
	Configured Antennas: TX 0x1 RX 0x1
	Supported interface modes:
		 * IBSS
		 * managed
		 * AP
		 * AP/VLAN
		 * monitor
		 * mesh point
		 * P2P-client
		 * P2P-GO
	Band 1:
		Capabilities: 0x116e
			HT20/HT40
			SM Power Save disabled
			RX HT20 SGI
			RX HT40 SGI
			RX STBC 1-stream
			Max AMSDU length: 3839 bytes
			DSSS/CCK HT40
		Maximum RX AMPDU length 65535 bytes (exponent: 0x003)
		Minimum RX AMPDU time spacing: 8 usec (0x06)
		HT TX/RX MCS rate indexes supported: 0-7
		Bitrates (non-HT):
			* 1.0 Mbps
			* 2.0 Mbps (short preamble supported)
			* 5.5 Mbps (short preamble supported)
			* 11.0 Mbps (short preamble supported)
			* 6.0 Mbps
			* 9.0 Mbps
			* 12.0 Mbps
			* 18.0 Mbps
			* 24.0 Mbps
			* 36.0 Mbps
			* 48.0 Mbps
			* 54.0 Mbps
		Frequencies:
			* 2412 MHz [1] (20.0 dBm)
			* 2417 MHz [2] (20.0 dBm)
			* 2422 MHz [3] (20.0 dBm)
			* 2427 MHz [4] (20.0 dBm)
			* 2432 MHz [5] (20.0 dBm)
			* 2437 MHz [6] (20.0 dBm)
			* 2442 MHz [7] (20.0 dBm)
			* 2447 MHz [8] (20.0 dBm)
			* 2452 MHz [9] (20.0 dBm)
			* 2457 MHz [10] (20.0 dBm)
			* 2462 MHz [11] (20.0 dBm)
			* 2467 MHz [12] (20.0 dBm)
			* 2472 MHz [13] (20.0 dBm)
			* 2484 MHz [14] (disabled)
	Supported commands:
		 * new_interface
		 * set_interface
		 * new_key
		 * start_ap
		 * new_station
		 * new_mpath
		 * set_mesh_config
		 * set_bss
		 * authenticate
		 * associate
		 * deauthenticate
		 * disassociate
		 * join_ibss
		 * join_mesh
		 * remain_on_channel
		 * set_tx_bitrate_mask
		 * frame
		 * frame_wait_cancel
		 * set_wiphy_netns
		 * set_channel
		 * set_wds_peer
		 * tdls_mgmt
		 * tdls_oper
		 * probe_client
		 * set_noack_map
		 * register_beacons
		 * start_p2p_device
		 * set_mcast_rate
		 * channel_switch
		 * Unknown command (104)
		 * connect
		 * disconnect
	Supported TX frame types:
		 * IBSS: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * managed: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * AP: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * AP/VLAN: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * mesh point: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-client: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-GO: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-device: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
	Supported RX frame types:
		 * IBSS: 0x40 0xb0 0xc0 0xd0
		 * managed: 0x40 0xd0
		 * AP: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * AP/VLAN: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * mesh point: 0xb0 0xc0 0xd0
		 * P2P-client: 0x40 0xd0
		 * P2P-GO: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * P2P-device: 0x40 0xd0
	software interface modes (can always be added):
		 * AP/VLAN
		 * monitor
	valid interface combinations:
		 * #{ managed, P2P-client } <= 2, #{ AP, mesh point, P2P-GO } <= 2,
		   total <= 2, #channels <= 1
	HT Capability overrides:
		 * MCS: ff ff ff ff ff ff ff ff ff ff
		 * maximum A-MSDU length
		 * supported channel width
		 * short GI for 40 MHz
		 * max A-MPDU length exponent
		 * min MPDU start spacing
	Device supports TX status socket option.
	Device supports HT-IBSS.
	Device supports SAE with AUTHENTICATE command
	Device supports low priority scan.
	Device supports scan flush.
	Device supports AP scan.
	Device supports per-vif TX power setting
	Driver supports a userspace MPM

**Comment by blackHatMonkey** [comment]
As @strasharo have already said this
```bash
sudo python wifiphisher.py -aI wlan1 -jI wlan0
```
should solve the problem.

**Comment by caixiawang** [comment]
Command  Python wifiphisher.py -aI wlan1 -jI wlan0 sudo  Not solve the problem
kali version
root@kali:~# uname -a
Linux kali 4.3.0-kali1-amd64 #1 SMP Debian 4.3.3-5kali4 (2016-01-13) x86_64 GNU/Linux

**Comment by caixiawang** [comment]
Can interfere with the user can not create fake access point

**Comment by caixiawang** [comment]
Hi @strasharo    Did you run successfully?

**Comment by caixiawang** [comment]
![GitHub Logo](https://cloud.githubusercontent.com/assets/16268280/12005415/c43359a2-abe7-11e5-8fa7-7123d7297e10.png)Borrowing a friend's image,can not create a fake access point,I really do not know how to solve the problem?

**Comment by strasharo** [comment]
Yes, I have been running it for ages without issues, just choose your Atheros adapter as the AP interface and the Realtek one for jamming.

**Comment by caixiawang** [comment]
![A](https://github.com/caixiawang/new/blob/master/image/4.png?raw=true)Why I can't create a fake access point, I really do not know how to solve this problem?

**Comment by caixiawang** [comment]
![gif](https://github.com/caixiawang/new/blob/master/image/1.gif?raw=true)

**Comment by naflan121** [comment]
does TL-WN722N actually support ap mode anyone who finish the attack successfully ?

**Comment by caixiawang** [comment]
I did not succeed, did not create the false AP

**Comment by blackHatMonkey** [comment]
@caixiawang 
Can you try creating a access point using the fallowing method:

1. Save the code below as ```hostapd.conf```
```
interface=wlan1
driver=nl80211
ssid=Test
hw_mode=g
channel=1
````
2. Run
```bash
hostapd hostapd.conf
```
This should create a access point named ```Test```. Please post if you got any erros .

**Comment by caixiawang** [comment]
![png](https://github.com/caixiawang/wifiphisher/blob/master/image/2.png?raw=true)
After running mistakes, to get to know wifiphisher recent software work, really a great effort

**Comment by caixiawang** [comment]
![png](https://github.com/caixiawang/wifiphisher/blob/master/image/2.png?raw=true)
After running mistakes

**Comment by caixiawang** [comment]
![png](https://github.com/caixiawang/wifiphisher/blob/master/image/2.png?raw=true)
Run Error

**Comment by blackHatMonkey** [comment]
It seems that there is a problem with your version of ```hostapd```. You have to remove it using this command:
```bash
sudo apt-get purge hostapd
```
And re-install it using:
```bash
sudo apt-get install hostapd
```
Tell us if that solved it.

**Comment by caixiawang** [comment]
Setting hostapd (1:2.3-2.3)...
update-rc.d: We have no instructions for the hostapd init script.
update-rc.d: It looks like a network service, we disable it.
insserv: warning: current start runlevel(s) (empty) of script `hostapd' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `hostapd' overrides LSB defaults (0 1 6).
Processing for systemd (228-4) triggers ...
root@kali:~# 
Run after unloading heavy equipment, still can not create an access point, hostapd seems to be prompted to reinstall a mistake?

**Comment by caixiawang** [comment]
![png](https://github.com/caixiawang/wifiphisher/blob/master/image/4.png?raw=true"Unable to create AP")
![png](https://github.com/caixiawang/wifiphisher/blob/master/image/3.png?raw=true"Unable to create AP")
uninitialized?

**Comment by caixiawang** [comment]
root@kali:~# nmcli radio wifi off
root@kali:~# sudo ifconfig wlan0 up
SIOCSIFFLAGS: Operation not possible due to RF-kill
sudo ifconfig wlan0 up command can not be executed, the teacher how to solve this problem?

**Comment by caixiawang** [comment]
![png](https://github.com/caixiawang/wifiphisher/blob/master/image/8.png?raw=true)
sudo ifconfig wlan0 up command can not be executed, the teacher how to solve this problem?

**Comment by blackHatMonkey** [comment]
You can use
```bash
sudo rfkill unblock wlan
```
to remove any soft block. Alternatively you can remove and re-insert the adapter and it should solve it.

**Comment by caixiawang** [comment]
![png](https://github.com/caixiawang/wifiphisher/blob/master/image/7.png?raw=true)
Forgive my understanding, Run, or the same mistakes?

**Comment by blackHatMonkey** [comment]
Can you run
```bash
rfkill list
```
and post the result.

**Comment by caixiawang** [comment]
![png](https://github.com/caixiawang/wifiphisher/blob/master/image/9.png?raw=true)
I do not know where the error?

**Comment by blackHatMonkey** [comment]
When the device is ```Hard blocked``` that means that the device is physically blocked by either a key(switch) or by system. If it is blocked by the system unplugging and re-inserting it should reset it it, otherwise, you have to find the switch to enable it.

**Comment by caixiawang** [comment]
Did you mean reseat the device?

**Comment by caixiawang** [comment]
Did you mean to re-disconnect device connections?

**Comment by caixiawang** [comment]
Contacting kali system soon, please enlighten

**Comment by caixiawang** [comment]
I don't understand what the teacher is saying? Reconnect the device

**Comment by blackHatMonkey** [comment]
I meant to disconnect the usb cable and reconnect it. After that check if the ```Hard blocked``` is set to ```No```.

**Comment by caixiawang** [comment]
root@kali:~# sudo rfkill list all
0: phy0: Wireless LAN
	Soft blocked: yes
	Hard blocked: no
1: phy1: Wireless LAN
	Soft blocked: no
	Hard blocked: no
Still not been resolved? I am a novice, I do not understand

**Comment by blackHatMonkey** [comment]
When you have
```
0: phy0: Wireless LAN
Soft blocked: yes
Hard blocked: no
```
this
```bash
sudo rfkill unblock all
```
should remove the ```soft block```. So if you run
```bash
rfkill list
```
you should get
```
0: phy0: Wireless LAN
Soft blocked: no
Hard blocked: no
```

**Comment by caixiawang** [comment]
Several twists and turns, the problem has been resolved! thank you very much

**Comment by caixiawang** [comment]
For another wireless card, wifiphisher run successfully, but the connection fake AP, suggesting that the site security certificate problem? Close program attack again. Tip
Unable to start HTTP server!
[-] Another process is running on port 8080.
[!] Closing

**Comment by sophron** [comment]
@caixiawang The certificate needs to be self-signed.  The tool output is also self-explanatory.

Note that we are not here to teach you WiFi hacking and this is not a support forum. Open an issue only if you encounter a bug that has not been reported already.

**Comment by caixiawang** [comment]
Still did not solve the problem?
root@kali:~/wifiphisher# sudo python wifiphisher.py -aI wlan1 -jI wlan0
[-] Unable to start HTTP server!
[-] Another process is running on port 8080.
[!] Closing

**Comment by strasharo** [comment]
You have another process listening on 8080. 
`netstat -ntpl | grep 8080`

**Comment by caixiawang** [comment]
Every time this happens, I need to restart KALI, there are other ways you? Wifiphisher unsuccessful attacks, how to stop?

**Comment by blackHatMonkey** [comment]
Try to list the process that is using that port with this command:
```bash
sudo fuser 8080/tcp
```
then when you find it's id, kill it:
```bash
sudo kill pid
```

**Comment by caixiawang** [comment]
Thank you, teacher, problem solved!

**Comment by caixiawang** [comment]
Teacher in trouble again, I re-purchase a separate wireless card, how do I know whether it supports packet injection it? What orders? Can you teach me?

**Comment by caixiawang** [comment]
Teacher in trouble again, I re-purchase a separate wireless card, how do I know whether it supports packet injection it? Can you teach me

**Comment by caixiawang** [comment]
Teacher in trouble again, I re-purchase a separate wireless card, how do I know whether it supports packet injection it? What is the command? Can you teach me?

**Comment by blackHatMonkey** [comment]
Run this:
```bash
iw phy phy0 info
```
and look into the supported interface modes. If you see ```monitor``` it means that it supports packet injection and if you see ```AP``` it means that ```AP``` mode is supported.

**Comment by caixiawang** [comment]
Many thanks! Wifiphisher hoped to do more powerful. I will always support you!

**Comment by caixiawang** [comment]
Today received TL-WN821N ver: 5.0 wireless card immediately test, kali is not supported! depressed

**Comment by djdisikd** [comment]
Can I know whether Alfa AWUS1900 supports ap mode?

**Comment by Leon-79** [comment]
Hello, I'm following all the threads on this forum but I cannot make wifiphisher work. I have an internal wifi device that doesn't support monitor mode, then I bought Alfa awus036nha that supports Mon and AP. I have a doubt now reading this thread, do I need 2 devices that support monitor mode??? For this I cannot connect my phone to my AP? I'm desperate.

**Comment by giveplease** [comment]
> I used two rtl8187 and TL-WN722N card, can not create a false ap? please help me

did you any solution for AP mode ?
